src/Microsoft.Health.Fhir.Web/Startup.cs

```diff
@@ -24,9 +25,16 @@ public virtual void ConfigureServices(IServiceCollection services)
         {
             services.AddDevelopmentIdentityProvider(Configuration);
 
-            services.AddFhirServer(Configuration)
-                .AddExportWorker()
-                .AddCosmosDb(Configuration);
+            Core.Registration.IFhirServerBuilder fhirServerBuilder = services.AddFhirServer(Configuration).AddExportWorker();
+
+            if (Configuration["DataStore"].Equals(KnownDataStores.CosmosDb, StringComparison.InvariantCultureIgnoreCase))
+            {
+                fhirServerBuilder.AddCosmosDb(Configuration);
+            }
+            else if (Configuration["DataStore"].Equals(KnownDataStores.SqlServer, StringComparison.InvariantCultureIgnoreCase))
+            {
+                services.AddExperimentalSqlServer();
```

Comment:
any reason this is not following the pattern above?

Reply:
The pattern of using the `FhirServerBuilder`?